packages/expo-dev-menu/README.md

```diff
@@ -1,26 +1,291 @@
-# expo-dev-menu
+# 📦 expo-dev-menu
 
 Expo/React Native module with the developer menu.
 
-# API documentation
+# ⚙️ Installation
 
-- [Documentation for the master branch](https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/docs.expo.io.md)
-- [Documentation for the latest stable release](https://docs.expo.io/versions/latest/sdk/docs.expo.io/)
+Firstly, you need to add the `expo-dev-menu` package to your project.
 
-# Installation in managed Expo projects
+<details>
+<summary>yarn</summary>
 
-For managed [managed](https://docs.expo.io/versions/latest/introduction/managed-vs-bare/) Expo projects, please follow the installation instructions in the [API documentation for the latest stable release](#api-documentation). If you follow the link and there is no documentation available then this library is not yet usable within managed projects &mdash; it is likely to be included in an upcoming Expo SDK release.
-
-# Installation in bare React Native projects
+```bash
+yarn add expo-dev-menu
+```
 
-For bare React Native projects, you must ensure that you have [installed and configured the `react-native-unimodules` package](https://github.com/expo/expo/tree/master/packages/react-native-unimodules) before continuing.
+</details>
 
-### Add the package to your npm dependencies
+<details>
+<summary>npm</summary>
 
-```
+```bash
 npm install expo-dev-menu
 ```
 
-# Contributing
+</details>
+
+Then you can start to configure the native projects using steps below.
+
+## 🤖 Android
+
+1.  Add the `dev-menu` to the android project.
+
+    I. Open `settings.gradle` and add the following lines:
+
+    ```gradle
+    include(":expo-dev-menu")
+    project(":expo-dev-menu").projectDir = new File("../node_modules/expo-dev-menu/android")
+
+    include(":expo-dev-menu-interface")
+    project(":expo-dev-menu-interface").projectDir = new File("../node_modules/expo-dev-menu-interface/android")
+    ```
+
+    II. Go to the `build.gradle` of your application and add `expo-dev-menu` as a dependency:
+
+    ```gradle
+    dependencies {
+        ...
+        implementation project(":expo-dev-menu-interface")
+        implementation project(":expo-dev-menu")
+        ...
+    }
+    ```
+
+    > _Note_: You don't have to use `implementationDebug` to add `expo-dev-menu` only to the debug builds. This package will be removed from the release build automatically.
+
+2.  Set up the `DevMenuManger` in the native code.
+
+    You can do it in two ways. We recommend using the basic initialization. However, if you have the custom activity in your application, then the advance one will be more suitable for you.
+
+    <details>
+    <summary>basic</summary>
+
+    Open the `MainActivity.java` or `MainActivity.kt` and make sure that your main activity class extends the `DevMenuAwareReactActivity`.
+
+      <details>
```

Comment:
Looks like these spoilers are not being nested as you probably wanted to do. Maybe better to have four `<details>`:
Kotlin (basic), Java (basic), Kotlin (advanced), Java (advanced)
or nest them by basic/advanced headers 🤔